Source/Urho3D/UI/Selectable.h

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright (c) 2008-2017 the Urho3D project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+#include "../UI/UIElement.h"
+
+namespace Urho3D
+{
+
+struct URHO3D_API Selectable : UIElement
+{
+    URHO3D_OBJECT(Selectable, UIElement);
+
+    using UIElement::UIElement;
+    virtual ~Selectable() = default;
+
+    Color selectionColor_ = Color(0.f, 0.f, 0.f, 0.f);
```

Comment:
Actually you never hide member variables from public access by default - virtually all classes in Urho start with `public` access specifier. using `struct` instead of `class` reduces redundant boilerplate code by saving the need to use `public` twice: before the base class and at the beginning of the class body.

Reply:
But they are declared in the private/protected part, yes?

Reply:
Ah, you mean the color variables. I overlooked it with the getters/setters. I'll fix it.

Reply:
Yes, I meant the variables. As we all know,`struct` or `class` are almost the same except for its default visibility. However, we also have to look at the existing code convention. I believe it would become a pitfall if we allow a single class hierarchy, like UIElement, has a mixed of `class` and `struct`.

Reply:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-struct

> Use class if the class has an invariant; use struct if the data members can vary independently